lightning/src/routing/router.rs

```diff
@@ -815,12 +829,11 @@ where L::Target: Logger {
 	let recommended_value_msat = final_value_msat * ROUTE_CAPACITY_PROVISION_FACTOR as u64;
 	let mut path_value_msat = final_value_msat;
 
-	// We don't want multiple paths (as per MPP) share liquidity of the same channels.
-	// This map allows paths to be aware of the channel use by other paths in the same call.
-	// This would help to make a better path finding decisions and not "overbook" channels.
-	// It is unaware of the directions (except for `next_outbound_htlc_limit_msat` in
-	// `first_hops`).
-	let mut bookkept_channels_liquidity_available_msat = HashMap::with_capacity(network_nodes.len());
+	// Keep track of how much liquidity has been used in selected channels. Used to determine
+	// if the channel can be used by additional MPP paths or to inform path finding decisions. It is
+	// aware of direction, but liquidity used in one direction will not offset any used in the
+	// opposite direction. Knowledge of direction ensures using the correct htlc_maximum_msat value.
+	let mut used_channel_liquidities = HashMap::with_capacity(network_nodes.len());
```

Comment:
I think annotating the type here would be really helpful, because it's implicit in all the following code. Particularly the tuple key access.